src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -0,0 +1,1931 @@
+            C.R(a.M);
+            C.R(x => a.M(x));
+
+            static class C
+            {
+                public static int M(this Span<int> x, int y) => x[y];
+                public static void R(Func<int, int> f) => Console.Write(f(1));
+            }
+            """;
+
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular12).VerifyDiagnostics(
+            // (4,5): error CS1503: Argument 1: cannot convert from 'method group' to 'System.Func<int, int>'
+            // C.R(a.M);
+            Diagnostic(ErrorCode.ERR_BadArgType, "a.M").WithArguments("1", "method group", "System.Func<int, int>").WithLocation(4, 5),
+            // (5,10): error CS1929: 'int[]' does not contain a definition for 'M' and the best extension method overload 'C.M(Span<int>, int)' requires a receiver of type 'System.Span<int>'
+            // C.R(x => a.M(x));
+            Diagnostic(ErrorCode.ERR_BadInstanceArgType, "a").WithArguments("int[]", "M", "C.M(System.Span<int>, int)", "System.Span<int>").WithLocation(5, 10));
+
+        var expectedDiagnostics = new[]
+        {
+            // (4,5): error CS1113: Extension method 'C.M(Span<int>, int)' defined on value type 'Span<int>' cannot be used to create delegates
+            // C.R(a.M);
+            Diagnostic(ErrorCode.ERR_ValueTypeExtDelegate, "a.M").WithArguments("C.M(System.Span<int>, int)", "System.Span<int>").WithLocation(4, 5)
+        };
+
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.RegularNext).VerifyDiagnostics(expectedDiagnostics);
+        CreateCompilationWithSpan(source).VerifyDiagnostics(expectedDiagnostics);
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void Conversion_Array_Span_Implicit_MethodGroup_FunctionPointer(LanguageVersion langVersion)
+    {
+        var source = """
+            using System;
+
+            unsafe
+            {
+                C.R(&C.M);
+            }
+
+            unsafe static class C
+            {
+                public static void M(Span<int> x) => Console.Write(x[1]);
+                public static void R(delegate*<int[], void> a) => a(new int[] { 1, 2, 3 });
+            }
+            """;
+
+        CreateCompilationWithSpan(source, TestOptions.UnsafeReleaseExe, TestOptions.Regular.WithLanguageVersion(langVersion)).VerifyDiagnostics(
+            // (5,9): error CS8757: No overload for 'C.M(Span<int>)' matches function pointer 'delegate*<int[], void>'
+            //     C.R(&C.M);
+            Diagnostic(ErrorCode.ERR_MethFuncPtrMismatch, "&C.M").WithArguments("C.M(System.Span<int>)", "delegate*<int[], void>").WithLocation(5, 9));
+    }
+
+    [Theory, CombinatorialData]
+    public void Conversion_Array_ReadOnlySpan_Covariant(bool cast)
+    {
+        var source = $$"""
+            using System;
+
+            class C
+            {
+                ReadOnlySpan<object> M(string[] x) => {{(cast ? "(ReadOnlySpan<object>)" : "")}}x;
+            }
```

Comment:
>} [](http://example.com/codeflow?start=12&length=1)

Are we testing contra-variance (should fail), and co- and contra-variance for `Span<T>` (should fail)?

Perhaps test:
```csharp
class C<T, U>
{
    ReadOnlySpan<U> F1(T[] x) => x;
    ReadOnlySpan<U> F2(T[] x) => (ReadOnlySpan<U>)x;
    Span<U> F3(T[] x) => x;
    Span<U> F4(T[] x) => (Span<U>)x;
}
```
... with:
- *no constraints*
- `where U : T`
- `where T : class where U : T, class`
- `where T : class, U where U : class`